video/out/drm_common.c

```diff
@@ -1240,6 +1353,52 @@ double vo_drm_get_display_fps(struct vo_drm_state *drm)
     return mode_get_Hz(&drm->mode.mode);
 }
 
+bool vo_drm_set_hdr_metadata(struct vo *vo, bool force_sdr)
+{
+    struct vo_drm_state *drm = vo->drm;
+    struct mp_image_params target_params = vo_get_target_params(vo);
+    if (!force_sdr && (pl_color_space_equal(&target_params.color, &drm->target_params.color) ||
+        !target_params.w || !target_params.h))
+        return false;
+
+    destroy_hdr_blob(drm);
+    drm->target_params = target_params;
+    bool use_sdr = !target_params_supported_by_display(drm) || force_sdr;
+
+    // For any HDR, the BT2020 drm colorspace is the only one that works in practice.
+    struct drm_atomic_context *atomic_ctx = drm->atomic_context;
+    int colorspace = !force_sdr && pl_color_space_is_hdr(&drm->target_params.color) ?
```

Comment:
@Dudemanguy

I took a closer look.

`pl_color_space_is_hdr()` is defined in libplacebo as:

``` c
bool pl_color_space_is_hdr(const struct pl_color_space *csp)
{
    return csp->hdr.max_luma > PL_COLOR_SDR_WHITE ||
           pl_color_transfer_is_hdr(csp->transfer);
}
```

`PL_COLOR_SDR_WHITE` is 203. So that function will return true if you simply set `target-peak` to any value > 203, even with SDR files. Playing a tone-mapped HDR file, which usually involves setting target-peak to values like 400, will also make that function return true.

That was the source of confusion for me. Maybe `pl_color_transfer_is_hdr()` should be used instead here.

I get broken rendering on my device when  "Colorspace" is set to any of

```
    DRM_MODE_COLORIMETRY_OPRGB = 7,
    DRM_MODE_COLORIMETRY_BT2020_RGB = 9,
    DRM_MODE_COLORIMETRY_BT2020_YCC = 10,
```

This is true even without switching to HDR mode. EDID info mentions support for the later two only.

No issue when "Colorspace" is explicitly set to `DRM_MODE_COLORIMETRY_DEFAULT` or `DRM_MODE_COLORIMETRY_BT709_YCC`. Other values don't apply.

Reply:
> But if your display advertises support for BT2020_RGB and it doesn't actually work, not sure what we can do?

Yeah, I'm aware the problem lies elsewhere, not with mpv. Just reporting my testing results.

In fact, I'm happy to report that I tested with my laptop and the same TV (same input and same HDMI wire too), and got HDR mode enabled with no broken rendering.

For reference, laptop's GPU is a Turing GTX 1650, and I use the nvidia driver (560.35.03).

So now we know that the issue is probably caused by a bug in the open AMD driver stack (POLARIS10 generation at least).